web/html/src/manager/appstreams/utils.ts

```diff
@@ -0,0 +1,46 @@
+import { AppStreamModule, Channel, ChannelAppStream } from "./appstreams.type";
+
+export const getStreamName = (module: AppStreamModule) => `${module.name}:${module.stream}`;
+
+export const handleModuleEnableDisable = (
+  channel: Channel,
+  appStream: AppStreamModule,
+  appStreams: ChannelAppStream[],
+  toEnable: Map<number, string[]>,
+  toDisable: Map<number, string[]>,
+  setToEnable: React.Dispatch<React.SetStateAction<Map<number, string[]>>>,
+  setToDisable: React.Dispatch<React.SetStateAction<Map<number, string[]>>>
+) => {
+  const handleReactStateChange = (prevState: Map<number, string[]>, stream: string) => {
+    const updatedMap = new Map<number, string[]>(prevState);
+    let newList = updatedMap.get(channel.id) ?? [];
+    newList = newList.includes(stream) ? (newList = newList.filter((it) => it !== stream)) : newList.concat(stream);
```

Comment:
This is a bit confusing, is the inline assignment `(newList = ...)` necessary? It seems like it should not be, since we assign the result in the outer assign anyway?